atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableCreator.java

```diff
@@ -61,71 +64,65 @@ void createTables(Map<TableReference, byte[]> tableRefToMetadata) {
 
     private void createTable(TableReference tableRef, byte[] metadata, CassandraClient client) throws TException {
         CqlQuery query = constructQuery(tableRef, metadata);
-        client.execute_cql3_query(query, Compression.NONE, org.apache.cassandra.thrift.ConsistencyLevel.QUORUM);
+        client.execute_cql3_query(query, Compression.NONE, CassandraKeyValueServiceImpl.WRITE_CONSISTENCY);
     }
 
     private CqlQuery constructQuery(TableReference tableRef, byte[] metadata) {
-        StringBuilder queryBuilder = new StringBuilder();
-
         TableMetadata tableMetadata = CassandraKeyValueServices.getMetadataOrDefaultToGeneric(metadata);
         boolean appendHeavyReadLight = tableMetadata.isAppendHeavyAndReadLight();
-
-        Map<String, String> settings = new HashMap<>();
-        settings.put("bloom_filter_fp_chance", falsePositive(tableMetadata.hasNegativeLookups(), appendHeavyReadLight));
-        settings.put("caching", "'KEYS_ONLY'");
-        settings.put("compaction", getCompaction(appendHeavyReadLight));
-        settings.put("compression", getCompression(tableMetadata.getExplicitCompressionBlockSizeKB()));
-        settings.put("gc_grace_seconds", Integer.toString(config.gcGraceSeconds()));
-        settings.put("id", "'%s'");
-        settings.put("populate_io_cache_on_flush", cachePriorityIsHottest(tableMetadata));
-        settings.put("speculative_retry", "'NONE'");
-
-        queryBuilder.append("CREATE TABLE IF NOT EXISTS \"%s\".\"%s\""
-                + " ( " + ROW + " blob, " + COLUMN + " blob, " + TIMESTAMP + " bigint, " + VALUE + " blob, "
-                + "PRIMARY KEY (" + ROW + ", " + COLUMN + ", " + TIMESTAMP + ")) "
-                + "WITH COMPACT STORAGE ");
-
-        settings.forEach((option, value) -> queryBuilder.append("AND " + option + " = " + value + " "));
-
-        queryBuilder.append("AND CLUSTERING ORDER BY (" + COLUMN + " ASC, " + TIMESTAMP + " ASC) ");
+        String keyspace = wrapInQuotes(config.getKeyspaceOrThrow());
+        String internalTableName = wrapInQuotes(CassandraKeyValueServiceImpl.internalTableName(tableRef));
+
+        String query = SchemaBuilder.createTable(keyspace, internalTableName)
+                .ifNotExists()
+                .addPartitionKey(ROW, DataType.blob())
+                .addClusteringColumn(COLUMN, DataType.blob())
+                .addClusteringColumn(TIMESTAMP, DataType.bigint())
+                .addColumn(VALUE, DataType.blob())
+                .withOptions()
+                .bloomFilterFPChance(falsePositive(tableMetadata.hasNegativeLookups(), appendHeavyReadLight))
+                .caching(SchemaBuilder.Caching.KEYS_ONLY)
+                .comment("")
```

Comment:
I think we can delete this

Reply:
I included it for completeness sake, but I guess it is slightly overkill :)